### PR TITLE
Add origin to signature request confirmation page

### LINKED ISSUE #6071: Signature request screen must show requesting domain
Currently our signature challenge screen as presented by 3Box does not show the requesting domain, so if I'm looking at another domain while the challenge pops up, I might think it's coming from them, which could be used to mitm the 3box secret material.

### PR DESCRIPTION
**Pull request by Gudahtt**
Fixes #6071

The origin of the dapp that suggested signing has been added to the signature request confirmation page. This only applies to `eth_sign`, `personal_sign`, `eth_signTypedData`, and `eth_signTypedData_v1`. The confirmation page for `eth_signTypedData_v3` and `eth_signTypedData_v4` already featured the origin.

<details>
<summary>Screenshots</summary>

`eth_sign`:

sig-req-origin

`eth_signTypedData`:

sig-req-typed

</details>

### CODE REVIEW

**Review by rachelcope — DISMISSED** [review]
Looks good for now as a quick fix

**Comment by rachelcope** [comment]
I'll be pushing for design refresh for this page soon 😸

File: ui/app/components/app/signature-request-original/signature-request-original.component.js [hunk 1]
```diff
@@ -148,11 +150,25 @@ export default class SignatureRequestOriginal extends Component {
...
+    const originMetadata = txData.msgParams.origin
+      ? domainMetadata?.[txData.msgParams.origin]
+      : null
+
     return (
-      <div className="request-signature__request-info">
-        <div className="request-signature__headline">
-          {this.context.t('yourSigRequested')}
+      <div className="request-signature__origin-row">
+        <div className="request-signature__origin-label">Origin:</div>
```
Comment (danjm) [inline ui/app/components/app/signature-request-original/signature-request-original.component.js:162]: Should "Origin:" be translated?

**Review by danjm — CHANGES_REQUESTED** [review]
Left one comment, otherwise code looks good

File: ui/app/components/app/signature-request-original/signature-request-original.component.js (hunk 1 above)
Comment (Gudahtt) [reply ui/app/components/app/signature-request-original/signature-request-original.component.js:162]: Of course :facepalm: Yes it should, good catch!

Comment (Gudahtt) [reply ui/app/components/app/signature-request-original/signature-request-original.component.js]: Resolved in a774097dd0c41cb4ec56ff1705ca966baf0803de

**Review by danjm — APPROVED** [review]
e2e fail seems to be unrelated. LGTM. Approved.

**Pull request merged**